friends/src/components/Login.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react'
+import { connect } from 'react-redux';
+import { login } from '../actions';
+
+class Login extends React.Component {
+    constructor() {
+        super();
+        this.state = {
+            credentials: {
+                username: '',
+                password: ''
+            }
+        }
+    }
+
+    handleChange = e => {
+        console.log(e.target.value)
+        this.setState({
+            credentials: {
+                ...this.state.credentials,
+            [e.target.name] : e.target.value
+            }
+        })
+    }
+
+    loggingIn = e => {
+        e.preventDefault();
+        this.props.login(this.state.credentials).then(()=>{
+            this.props.history.push('/friends')
+        })
+    }
+    render() {
+        return(
+            <div className="login">
+            <h1>Please Login</h1>
+                <form className="login-form" onSubmit={this.loggingIn}>
+                    <input onChange={this.handleChange} value={this.state.credentials.username} name="username" type="text" required />
+                    <input onChange={this.handleChange} type="password" value={this.state.credentials.password} name="password" required />
+                    <button>Login</button>
+                </form>
+            </div>
+        )
+    }
+}
+
+const mapStateToProps = state => ({
+    loading: state.loading
+});
+
```

Comment:
While utilizing a `loading` state variable is a great idea here, it is not used and it is also not declared in your reducer's `state` object.

You could use this `loading` variable (or something similarly named) for conditional rendering (to render some kind of placeholder component while an operation is still ongoing), provided that you defined it in your reducer's `state` object.